src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleEqualityTests.cs

```diff
@@ -810,13 +813,196 @@ static void Main()
+                .Where(e => e.Kind() == SyntaxKind.DefaultLiteralExpression);
+
+            foreach (var literal in defaultLiterals)
+            {
+                Assert.Equal("default", literal.ToString());
+                var info = model.GetTypeInfo(literal);
+                Assert.Equal("(System.String, System.String)?", info.Type.ToTestDisplayString());
+                Assert.Equal("(System.String, System.String)?", info.ConvertedType.ToTestDisplayString());
+            }
+        }
+
+        [Fact]
+        public void TestNullableTupleAndDefault_Nested()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (string, string)? t = (null, null);
+        System.Console.Write((null, t) == (null, default));
+        System.Console.Write((t, null) != (default, null));
+    }
+}";
+            var comp = CreateCompilation(source, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "FalseTrue");
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+
+            var defaultLiterals = tree.GetCompilationUnitRoot().DescendantNodes().OfType<LiteralExpressionSyntax>()
+                .Where(e => e.Kind() == SyntaxKind.DefaultLiteralExpression);
+
+            foreach (var literal in defaultLiterals)
+            {
+                Assert.Equal("default", literal.ToString());
+                var info = model.GetTypeInfo(literal);
+                Assert.Equal("(System.String, System.String)?", info.Type.ToTestDisplayString());
+                Assert.Equal("(System.String, System.String)?", info.ConvertedType.ToTestDisplayString());
+            }
+        }
+
+        [Fact]
+        public void TestNestedDefaultWithNonTupleType()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        System.Console.Write((null, 1) == (null, default));
+        System.Console.Write((0, null) != (default, null));
+    }
+}";
+            var comp = CreateCompilation(source, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "FalseFalse");
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+
+            var defaultLiterals = tree.GetCompilationUnitRoot().DescendantNodes().OfType<LiteralExpressionSyntax>()
+                .Where(e => e.Kind() == SyntaxKind.DefaultLiteralExpression);
+
+            foreach (var literal in defaultLiterals)
+            {
+                Assert.Equal("default", literal.ToString());
+                var info = model.GetTypeInfo(literal);
+                Assert.Equal("System.Int32", info.Type.ToTestDisplayString());
+                Assert.Equal("System.Int32", info.ConvertedType.ToTestDisplayString());
+            }
+        }
+
+        [Fact]
+        public void TestNestedDefaultWithNullableNonTupleType()
```

Comment:
>TestNestedDefaultWithNullableNonTupleType [](start = 20, length = 41)

Do we have a test where ```default``` is paired with ```null```?

Reply:
Yes, that's covered in `TestTypelessTupleAndTupleOfDefaults` at line 1041.
`System.Console.Write((null, () => 1) == (default, default));`